File: be/mytown/Models/DbModels/Institution.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace mytown.Models.DbModels
{
    public class Institution
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int? LogoID { get; set; } 
        public int HeadquaterID { get; set; }
        public string Address { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public DateTime JoinDate { get; set; }
        public string Password { get; set; }

        [ForeignKey("HeadquaterID")]
        public virtual City City { get; set; }

        [ForeignKey("LogoID")]
        public UserEntityImage Logo { get; set; }
        public virtual List<Post> Posts { get; set; }
        public virtual List<Comment> Comments{ get; set; }
        [ForeignKey("Id")]
        public virtual UserEntity UserEntity { get; set; }
    }
}

File: fe/web_app/lib/models/DbModels/institution.dart
class Institution
{
  String _name;
  int _headquaterID;
  String _email;
  String _password;
  String _address;
  String _phone;
  DateTime _joinDate;


  Institution(this._name, this._headquaterID, this._email, this._password, this._address, this._phone, this._joinDate);
  
  Map<String, dynamic> toMap(){
    var data  = Map<String, dynamic>();

    data["name"] = this._name;
    data["headquaterID"] = this._headquaterID;
    data["email"] = this._email;
    data["password"] = this._password;
    data["address"] = this._address;
    data["phone"] = this._phone;
    data["joinDate"] = this._joinDate.toIso8601String();

    return data;
  }
}